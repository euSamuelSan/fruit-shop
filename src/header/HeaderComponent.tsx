import React, { useState } from 'react'
import styled from 'styled-components'
import { AiOutlineShoppingCart, AiOutlineLogout } from 'react-icons/ai'
import { IoLeafOutline } from 'react-icons/io5'
import { Navigate } from 'react-router-dom'

import Colors from '../commons/Colors'
import { useUserContext } from '../context/UserContext'
import CheckoutModalComponent from '../shop/checkout/CheckoutModalComponent'

export default function HeaderComponent(): JSX.Element {
    const [isCheckoutModalVisible, setIsCheckoutModalVisible] =
        useState<boolean>(false)
    const [redirectToLogin, setRedirectToLogin] = useState<boolean>(false)
    const { cartItems } = useUserContext()

    const handleCartIndicator = () => {
        const itemsInCart = Object.keys(cartItems).length
        return itemsInCart ? (
            <CartCounterIndicator>
                <CartCounterText>{itemsInCart}</CartCounterText>
            </CartCounterIndicator>
        ) : null
    }

    const logout = () => {
        localStorage.clear()
        setRedirectToLogin(true)
    }

    return (
        <Container>
            <LeafIcon color={Colors.white} size={40} />
            <Title>Sua Horta</Title>
            <ActionsContainer>
                <Action onClick={() => setIsCheckoutModalVisible(true)}>
                    <>
                        <AiOutlineShoppingCart
                            color={Colors.white}
                            size={24}
                            title="Carrinho"
                        />
                        {handleCartIndicator()}
                    </>
                </Action>
                <Action onClick={() => logout()}>
                    <>
                        <AiOutlineLogout
                            color={Colors.white}
                            size={24}
                            title="Sair"
                        />
                        {redirectToLogin && <Navigate to="/" />}
                    </>
                </Action>
            </ActionsContainer>
            <CheckoutModalComponent
                isModalVisible={isCheckoutModalVisible}
                setIsModalVisible={setIsCheckoutModalVisible}
            />
        </Container>
    )
}

const Container = styled.div`
    height: 50px;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: ${Colors.leafyGreen};
    position: sticky;
    top: 0px;
    box-shadow: 0 1px 3px 2px rgba(0, 0, 0, 0.1);
`

const LeafIcon = styled(IoLeafOutline)`
    position: relative;
    left: 6px;
    margin-left: auto;
`

const Title = styled.h1`
    color: ${Colors.white};
    margin: 0;
`

const ActionsContainer = styled.div`
    display: flex;
    gap: 10px;
    margin-left: auto;
    padding-right: 20px;

    @media (max-width: 370px) {
        padding-right: 5px;
    }
`

const Action = styled.button`
    display: flex;
    border: none;
    background: none;
    cursor: pointer;
    justify-content: center;
    align-content: center;
    align-items: center;
    border-radius: 25px;

    width: 34px;
    height: 34px;

    &:hover {
        background-color: rgb(0, 0, 0, 0.1);
        transition: background-color 0.2s;
    }
`

const CartCounterIndicator = styled.div`
    width: 14px;
    height: 14px;
    border-radius: 25px;
    background-color: ${Colors.strawberry};
    position: absolute;
    top: 8px;
    right: 62px;
`
const CartCounterText = styled.p`
    color: ${Colors.white};
    margin: 0;
    padding-top: 2px;
    font-size: 10px;
`
