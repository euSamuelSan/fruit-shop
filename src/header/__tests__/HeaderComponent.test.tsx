import { render, RenderResult } from '@testing-library/react'

import HeaderComponent from '../HeaderComponent'

describe('HeaderComponent', () => {
    const renderHeaderComponent = (): RenderResult => {
        return render(<HeaderComponent />)
    }

    it('Deve desenhar o componente corretamente', () => {
        renderHeaderComponent()

        expect(document.body).toMatchSnapshot()
    })
})
