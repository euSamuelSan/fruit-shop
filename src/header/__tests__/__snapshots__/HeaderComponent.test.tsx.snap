// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent Deve desenhar o componente corretamente 1`] = `
<body>
  <div>
    <div
      class="sc-idiyUo wCFDq"
    >
      <svg
        class="sc-dIouRR cUxhAQ"
        color="#ffffff"
        fill="currentColor"
        height="40"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(255, 255, 255);"
        viewBox="0 0 512 512"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M321.89 171.42C233 114 141 155.22 56 65.22c-19.8-21-8.3 235.5 98.1 332.7 77.79 71 197.9 63.08 238.4-5.92s18.28-163.17-70.61-220.58zM173 253c86 81 175 129 292 147"
          fill="none"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="32"
        />
      </svg>
      <h1
        class="sc-hHLeRK Cxlpw"
      >
        Sua Horta
      </h1>
      <div
        class="sc-dmRaPn hECnYE"
      >
        <button
          class="sc-kgflAQ ksRWXD"
        >
          <svg
            color="#ffffff"
            fill="currentColor"
            height="24"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 1024 1024"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Carrinho
            </title>
            <path
              d="M922.9 701.9H327.4l29.9-60.9 496.8-.9c16.8 0 31.2-12 34.2-28.6l68.8-385.1c1.8-10.1-.9-20.5-7.5-28.4a34.99 34.99 0 0 0-26.6-12.5l-632-2.1-5.4-25.4c-3.4-16.2-18-28-34.6-28H96.5a35.3 35.3 0 1 0 0 70.6h125.9L246 312.8l58.1 281.3-74.8 122.1a34.96 34.96 0 0 0-3 36.8c6 11.9 18.1 19.4 31.5 19.4h62.8a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7h161.1a102.43 102.43 0 0 0-20.6 61.7c0 56.6 46 102.6 102.6 102.6s102.6-46 102.6-102.6c0-22.3-7.4-44-20.6-61.7H923c19.4 0 35.3-15.8 35.3-35.3a35.42 35.42 0 0 0-35.4-35.2zM305.7 253l575.8 1.9-56.4 315.8-452.3.8L305.7 253zm96.9 612.7c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6zm325.1 0c-17.4 0-31.6-14.2-31.6-31.6 0-17.4 14.2-31.6 31.6-31.6s31.6 14.2 31.6 31.6a31.6 31.6 0 0 1-31.6 31.6z"
            />
          </svg>
        </button>
        <button
          class="sc-kgflAQ ksRWXD"
        >
          <svg
            color="#ffffff"
            fill="currentColor"
            height="24"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 1024 1024"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title>
              Sair
            </title>
            <path
              d="M868 732h-70.3c-4.8 0-9.3 2.1-12.3 5.8-7 8.5-14.5 16.7-22.4 24.5a353.84 353.84 0 0 1-112.7 75.9A352.8 352.8 0 0 1 512.4 866c-47.9 0-94.3-9.4-137.9-27.8a353.84 353.84 0 0 1-112.7-75.9 353.28 353.28 0 0 1-76-112.5C167.3 606.2 158 559.9 158 512s9.4-94.2 27.8-137.8c17.8-42.1 43.4-80 76-112.5s70.5-58.1 112.7-75.9c43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 7.9 7.9 15.3 16.1 22.4 24.5 3 3.7 7.6 5.8 12.3 5.8H868c6.3 0 10.2-7 6.7-12.3C798 160.5 663.8 81.6 511.3 82 271.7 82.6 79.6 277.1 82 516.4 84.4 751.9 276.2 942 512.4 942c152.1 0 285.7-78.8 362.3-197.7 3.4-5.3-.4-12.3-6.7-12.3zm88.9-226.3L815 393.7c-5.3-4.2-13-.4-13 6.3v76H488c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h314v76c0 6.7 7.8 10.5 13 6.3l141.9-112a8 8 0 0 0 0-12.6z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;
